### ISSUE TITLE
Hierarchical parameter of custom models' forward method

### ISSUE THREAD

**Issue by jan-fathomx** [keyu-tian/SparK#91]
Hi! I have a question about the custom forward method for models, which includes the `hierarchical` argument. In the documentation in `models/custom.py` it says that when `hierarchical==False`, the model should return the logits of the classification task. However, both the `models/resnet.py` and `models/convnext.py` implementations appear incorrect.

- In the former, the forward method skips layers 1-4, resulting in a channel mismatch when passing the output of `self.maxpool(x)` to the fully connected layer. 
- In the latter, the forward method method skips right to the fully connected layer, also resulting in a tensor sizer mismatch error

Is the hierarchical parameter even necessary? Is there a scenario in which it's used during the pretraining and if so, what should the actual behavior be?
